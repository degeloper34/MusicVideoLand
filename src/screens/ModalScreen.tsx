import {FontAwesome} from "@expo/vector-icons";
import {useContext} from "react";
import {FlatList, Pressable, StyleSheet, Text, View} from "react-native";
import {Genre, RootStackScreenProps} from "../../types";
import Colors from "../constants/Colors";
import {MainContext} from "../context/mainContext";

export default function ModalScreen({
  navigation,
  route,
}: RootStackScreenProps<"Modal">) {
  const {genreList} = route.params;

  const context = useContext(MainContext);

  const renderItem = ({item}: {item: Genre}) => {
    return (
      <Pressable
        onPress={() => {
          context?.setSelectedGenreId(item.id);
          navigation.pop();
        }}
        style={{
          flexDirection: "row",
          alignItems: "center",
          marginBottom: 10,
          flex: 1,
        }}
      >
        <View style={{padding: 5, marginRight: 5}}>
          <FontAwesome
            name={
              item.id === context?.selectedGenreId ? "check-circle" : "circle"
            }
            size={25}
            color={Colors.white}
          />
        </View>
        <Text style={{color: Colors.white}}>{item.name}</Text>
      </Pressable>
    );
  };

  const onPressClearFilter = () => {
    context?.setSelectedGenreId(-1);
    navigation.pop();
  };

  return (
    <View style={styles.container}>
      <FlatList
        data={genreList}
        renderItem={renderItem}
        numColumns={2}
        ListFooterComponent={
          <Pressable
            style={{
              alignItems: "center",
            }}
            onPress={onPressClearFilter}
          >
            <Text
              style={{color: Colors.white, textDecorationLine: "underline"}}
            >
              Clear Filter
            </Text>
          </Pressable>
        }
      />
    </View>
  );
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    justifyContent: "center",
    backgroundColor: Colors.black,
  },
});
