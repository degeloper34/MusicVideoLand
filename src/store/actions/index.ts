export * from "./musicVideoActions";
export * from "./searchActions";
