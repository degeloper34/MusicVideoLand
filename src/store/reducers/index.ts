import {combineReducers} from "redux";

import musicVideoReducer from "./musicVideoReducers";
import searchReducer from "./searchReducers";

export const rootReducer = combineReducers({
  musicVideoReducer,
  searchReducer,
});
