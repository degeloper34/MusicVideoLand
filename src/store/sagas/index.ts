import {all} from "redux-saga/effects";

import {musicVideoWatcher} from "./musicVideoSagas";

export default function* rootSaga() {
  yield all([musicVideoWatcher()]);
}
