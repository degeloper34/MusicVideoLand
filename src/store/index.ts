export type RootState = ReturnType<typeof store.getState>;
export type AppDispatch = typeof store.dispatch;

import {applyMiddleware, legacy_createStore as createStore} from "redux";
import createSagaMiddleware from "redux-saga";
import {rootReducer} from "./reducers";
import rootSaga from "./sagas";

const sagaMiddleware = createSagaMiddleware();

export const store = createStore(rootReducer, applyMiddleware(sagaMiddleware));

sagaMiddleware.run(rootSaga);
