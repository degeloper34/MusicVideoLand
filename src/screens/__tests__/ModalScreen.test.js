import renderer from "react-test-renderer";
import React from "react";
import ModalScreen from "../ModalScreen";
import {Provider} from "react-redux";
import {store} from "../../store";

describe("ModalScreen Tests", () => {
  test("ModalScreen | Render without issues", () => {
    const screen = renderer
      .create(
        <Provider store={store}>
          <ModalScreen />
        </Provider>
      )
      .toJSON();

    expect(screen).toMatchSnapshot();
  });
});
