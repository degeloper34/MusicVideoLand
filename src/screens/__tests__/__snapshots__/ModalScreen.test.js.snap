// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ModalScreen Tests ModalScreen | Render without issues 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#000",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <RCTScrollView
    ListHeaderComponent={
      <Memo(Pressable)
        onPress={[Function]}
      >
        <CustomText
          style={
            Object {
              "alignSelf": "flex-end",
              "marginBottom": 20,
            }
          }
          text="Clear Filter"
          textColor="#fff"
          type="bold"
          underline={true}
        />
      </Memo(Pressable)>
    }
    contentContainerStyle={
      Object {
        "padding": 20,
      }
    }
    data={Array []}
    getItem={[Function]}
    getItemCount={[Function]}
    initialNumToRender={10}
    keyExtractor={[Function]}
    onContentSizeChange={[Function]}
    onLayout={[Function]}
    onMomentumScrollBegin={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={50}
    showsVerticalScrollIndicator={false}
    stickyHeaderIndices={Array []}
    viewabilityConfigCallbackPairs={Array []}
  >
    <View>
      <View
        onLayout={[Function]}
      >
        <View
          accessible={true}
          collapsable={false}
          focusable={true}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
        >
          <Text
            allowFontScaling={false}
            style={
              Array [
                Object {
                  "alignSelf": "flex-end",
                  "marginBottom": 20,
                },
                Object {
                  "color": "#fff",
                  "fontFamily": "sora-bold",
                  "fontSize": 12,
                  "textDecorationLine": "underline",
                },
              ]
            }
          >
            Clear Filter
          </Text>
        </View>
      </View>
    </View>
  </RCTScrollView>
</View>
`;
